pants.ini

```diff
@@ -152,7 +152,7 @@ deps: ["3rdparty:thrift"]
 
 
 [gen.thrift-py]
-deps: ["3rdparty/python:thrift"]
+deps: ["examples/3rdparty/python:thrift"]
```

Comment:
Is there a way to specify a jvm tool-esque "use this python requirement, but you can override it with a target in `pants.ini`"? Put another way, would the example in `examples/tests/python/example_test/usethriftpy:usethrift` fail without specifying this? If we turned `deps` into being a `python_requirement_library()` (we would probably deprecate `deps` and use another option name) we could maybe have something like what jvm tools provide.

Reply:
> Put another way, would the example in examples/tests/python/example_test/usethriftpy:usethrift fail without specifying this?

Yes, I believe so. Not 100% certain though.

Reply:
Oh, but the JVM side actually works the same way as this. If you generate a scrooge codegen target, an actual scrooge 3rdparty dep must exist, in order to minimize the magic of completely undeclared deps ending up in your binary.

Only the _compiler_ itself (which you don't end up with a runtime dep on) can be defaulted the way you are describing.

Reply:
Ok, but that's a big undocumented barrier to entry to using scrooge and codegen in pants in general, which is one of the first things the site states as a reason to use pants. I understand the rationale of not wanting undeclared deps, but having the default classpath in jvm tools definitely matches my understanding of "declared", the only difference is that it is not a specific target in your repro. I think every user wanting to use codegen will accept that they will need to import a specific library with types and interfaces. I can make this into a separate issue.

Reply:
Made #7176.